common/mapper/src/main/java/io/helidon/common/mapper/BackedValue.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright (c) 2021 Oracle and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.helidon.common.mapper;
+
+import java.util.Objects;
+import java.util.Optional;
+import java.util.function.Function;
+
+class BackedValue<T> extends EmptyValue<T> {
```

Comment:
Why extend the EmptyValue? Only common func seems to be name accessor

Reply:
yes, that is the only common thing. On the other hand, why implement it twice? Something with value is a logical extension of something without a value...